lib/puppet/pops/issues.rb

```diff
@@ -490,6 +490,10 @@ def self.hard_issue(issue_code, *args, &block)
     _("Unacceptable location. The name '%{name}' is unacceptable in file '%{file}'") % { name: name, file: file }
   end
 
+  ILLEGAL_TOP_CONSTRUCT_LOCATION = issue :ILLEGAL_TOP_CONSTRUCT_LOCATION, name, :file do
+    _("Unacceptable location. Unacceptable top level construct %{name}in file '%{file}'") % { name: name, file: file }
```

Comment:
Change `%{name}in` to `%{name} in` (missing space).

I think that this will give you double output of file since you pass a "reference AST" to `accept` and its source location is used to determine file, line and position on line.